src/features/workout-session/ui/workout-session-heatmap.tsx

```diff
@@ -94,22 +126,26 @@ export const WorkoutSessionHeatmap: React.FC<Props> = ({
       const y = MONTH_LABEL_HEIGHT + (PANEL_SIZE + PANEL_MARGIN) * j;
       const numOfColors = panelColors.length;
       const color = contribution.value >= numOfColors ? panelColors[numOfColors - 1] : panelColors[contribution.value];
-      // TODO i18n
-      const d = dayjs(until, dateFormat)
+      const d = dayjs(until, ISO_DATE_FORMAT)
         .endOf("week")
         .subtract((columns - i - 1) * 7 + (6 - j), "day");
-      const dateStr = d.format(dateFormat);
-      const tooltip =
-        contribution.value > 0 ? (
-          <div className="text-xs text-slate-50">
-            {dateStr} : <br />
-            {contribution.value} workout{contribution.value > 1 ? "s" : ""}
-          </div>
-        ) : (
-          <div className="text-xs text-slate-50">
-            {dateStr} : <br /> No workout
-          </div>
-        );
+      const dateStr = formatDate(d);
+
+      // Create tooltip content based on workout count
+      const getWorkoutText = (count: number): string => {
+        if (count === 0) return t("heatmap.no_workout");
+        if (count === 1) return `${count} ${t("heatmap.one_workout_unit")}`;
+        return `${count} ${t("heatmap.multiple_workouts_unit")}`;
+      };
```

Comment:
When you implemented my first comment, then you'll need to do the same pattern as this : 
```
{t("programs.set", { count: totalSets })}
```